### PR TITLE
Check glide version

### PR DESCRIPTION
**Pull request by soltysh**
Same thing as in https://github.com/openshift/client-go/pull/19

/assign @deads2k @stevekuznetsov

### CODE REVIEW

**Comment by soltysh** [comment]
@sttts I've also addressed all the comments from other prs as well

File: hack/lib/util/golang.sh [hunk 1]
```diff
@@ -11,8 +11,7 @@ function os::golang::verify_go_version() {
 	if [[ "${go_version[2]}" < "${OS_REQUIRED_GO_VERSION}" ]]; then
 		os::log::info "Detected go version: ${go_version[*]}."
 		if [[ -z "${PERMISSIVE_GO:-}" ]]; then
-			os::log::warning "Please install Go version ${OS_REQUIRED_GO_VERSION} or use PERMISSIVE_GO=y to bypass this check."
-			return 0
```
Comment (sttts) [inline hack/lib/util/golang.sh]: intentional? Then also delete the return below.

File: hack/lib/util/golang.sh [hunk 2]
```diff
@@ -21,3 +20,17 @@ function os::golang::verify_go_version() {
...
 }
 readonly -f os::golang::verify_go_version
+
+# os::golang::verify_glide_version ensures the glide is at the required level
+function os::golang::verify_glide_version() {
+	os::util::ensure::system_binary_exists 'glide'
+
+	local glide_version
+	glide_version=($(glide --version))
+	if ! echo "${glide_version[2]#v}" | awk -F. -v min=$OS_GLIDE_MINOR_VERSION '{ exit $2 < min }'; then
```
Comment (sttts) [inline hack/lib/util/golang.sh]: Seems to do what you want, even on Mac, but it's ugly. And let's hope there is never a Glide 1.0.

File: hack/lib/util/golang.sh (hunk 1 above)
Comment (soltysh) [reply hack/lib/util/golang.sh]: Yes, this needs to fail, when you want this to fallback you need to use `PERMISSIVE_GO` envvar. Not sure which return you're talking about.

File: hack/lib/util/golang.sh (hunk 2 above)
Comment (soltysh) [reply hack/lib/util/golang.sh]: I know, but that's the best I could come up with :/ I hope that by then dep will be mature enough ;)

**Comment by soltysh** [comment]
/retest

On Wed, Jan 3, 2018 at 2:13 PM, OpenShift CI Robot <notifications@github.com

**Comment by soltysh** [comment]
The script is identical to the other PRs already merged in api and client-go that were already approved.

**Comment by dmage** [comment]
AFAIK, the master copy of these files are supposed to be at https://github.com/openshift/release/tree/master/tools/hack/golang
/cc @stevekuznetsov

**Pull request merged**